src/_pytest/_code/code.py

```diff
@@ -704,7 +704,12 @@ def match(self, regexp: Union[str, Pattern[str]]) -> "Literal[True]":
         If it matches `True` is returned, otherwise an `AssertionError` is raised.
         """
         __tracebackhide__ = True
-        value = str(self.value)
+        value = "\n".join(
```

Comment:
We might want to work with @agronholm to ensure note helpers in the backport as well as shared formatting as follow up

Reply:
The tests here don't seem to involve exception groups though.

Reply:
@agronholm my impression is that exception notes are closely related and my understanding is that integration with the backport may be helpful as per https://github.com/agronholm/exceptiongroup/issues/31 and since formatting may also be affected my impression is that the most benefit for library users would come from having one integration place instead of multiple